java/memory/memory-core/src/main/java/org/apache/arrow/memory/util/StackTrace.java

```diff
@@ -29,6 +29,7 @@ public class StackTrace {
   /**
    * Constructor. Captures the current stack trace.
    */
+  @SuppressWarnings("nullness:assignment") //incompatible types in assignment
```

Comment:
Can you expand on why this is suppressed? `nullness:assignment` warning makes me think we can just check the length of the stack trace to make sure we are copying valid entries.

Reply:
> Can you expand on why this is suppressed? `nullness:assignment` warning makes me think we can just check the length of the stack trace to make sure we are assigning valid entries.

It is not necessary, I will update it.